src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -91,8 +96,77 @@ public void failed(Throwable exc, Void v) {
         }
     }
 
-    /** Returns a file object for the given cache key. */
-    File getFileForKey(String key) {
-        return new File(mRootDirectorySupplier.get(), DiskBasedCacheUtility.getFilenameForKey(key));
+    /** Puts the cache entry with a specified key into the cache. */
+    @Override
+    public void put(final String key, Cache.Entry entry, final OnPutCompleteCallback callback) {
+        // If adding this entry would trigger a prune, but pruning would cause the new entry to be
+        // deleted, then skip writing the entry in the first place, as this is just churn.
+        // Note that we don't include the cache header overhead in this calculation for simplicity,
+        // so putting entries which are just below the threshold may still cause this churn.
+        if (mTotalSize + entry.data.length > mMaxCacheSizeInBytes
+                && entry.data.length
+                        > mMaxCacheSizeInBytes * DiskBasedCacheUtility.HYSTERESIS_FACTOR) {
+            return;
+        }
+        final File file = DiskBasedCacheUtility.getFileForKey(key, mRootDirectorySupplier);
+        Path path = Paths.get(file.getPath());
+
+        try (AsynchronousFileChannel afc =
+                AsynchronousFileChannel.open(path, StandardOpenOption.WRITE)) {
+            final CacheHeader header = new CacheHeader(key, entry);
+            int headerSize = header.getHeaderSize();
+            int size = entry.data.length + headerSize;
+            ByteBuffer buffer = ByteBuffer.allocate(size);
+            header.writeHeader(buffer);
+            buffer.put(entry.data);
+            afc.write(
+                    /* source= */ buffer,
+                    /* position= */ 0,
+                    /* attachment= */ null,
+                    new CompletionHandler<Integer, Void>() {
+                        @Override
+                        public void completed(Integer integer, Void aVoid) {
```

Comment:
could you find a better name for `integer` and `aVoid` to be more descriptive? As is, I don't know what they stand for when reading the code.

If they're ignored, you can name them `ignore`, but prefer a descriptive name, especially for the integer.

Reply:
I'll change these